src/main/java/net/dv8tion/jda/api/events/guild/member/update/GuildMemberUpdateNicknameEvent.java

```diff
@@ -30,17 +31,17 @@
  *
  * <p>Identifier: {@code nick}
  */
-public class GuildMemberUpdateNicknameEvent extends GenericGuildMemberEvent implements UpdateEvent<Member, String>
+public class GuildMemberUpdateNicknameEvent extends GenericGuildMemberUpdateEvent<String>
 {
     public static final String IDENTIFIER = "nick";
 
     private final String oldNick, newNick;
 
     public GuildMemberUpdateNicknameEvent(@Nonnull JDA api, long responseNumber, @Nonnull Member member, @Nullable String oldNick)
     {
-        super(api, responseNumber, member);
+        super(api, responseNumber, member, oldNick, member.getNickname(), IDENTIFIER);
         this.oldNick = oldNick;
-        this.newNick = member.getNickname();
+        this.newNick = super.getNewValue();
     }
 
```

Comment:
All methods besides `getOldNickname` and `getNewNickname` can be removed.

Reply:
Alright